### DISCUSSION TITLE
Run docker with HTTPS

### DISCUSSION THREAD

**Discussion by dimdenGD** [coder/code-server#1048] [Q&A] [answered]
Hi!

I'm total noob in docker, so I want to ask a question.

I know how to run code-server with cert and cert-key without docker, but how to do it with docker?

I tried this:
```
docker run -it -p 127.0.0.1:8080:8080 -v "./cs/code-server --cert [certpath] --cert-key [privatekey]" -v "./project" codercom/code
-server:v2
```

But it runs server without HTTPS.

Thanks

**Comment by sr229** [comment]
Heya @dimdenGD, make sure you mounted your certificates first, then invoke your certificates in the flags. You can use LE if you don't want to self-sign certificates, I think we outlined it pretty well on the quickstart.

**Comment by dimdenGD** [comment]
How do I mount my certificates? I was reading docker docs but I didn't understood anything ;-;

Also how do I change Dockerfile

**Comment by sr229** [comment]
it should be easy as `docker -v /path/to/certs.crt:/home/coder/cert.crt -v /path/to/cert-key.key:/home/coder/cert.key -p 8080:8080 run codercom/code-server:v2 --cert /home/coder/cert.crt --cert-key /home/coder/cert.key` if I remember how to do volumes right.

**Comment by dimdenGD** [comment]
Now i'm getting `error EACCES: permission denied, open '/home/coder/cert.key'`

**Comment by mateothegreat** [comment]
@dimdenGD you should use a reverse proxy so you don't have to expose your private key to the running container. You can use something like HAproxy or nginx even. Let me know if you need help!

**Comment by demyxco** [comment, marked as answer]
@dimdenGD check out my docker-compose.yml on my [repo](https://github.com/demyxco/code-server/), it has a configured reverse proxy for SSL using just environment variables. Just remove the comments (#).

**Reply by LostOnTheLine** [reply]
Is this still not a thing? the crappy password-only login is not very secure, even under SSL, & even in a home lab you want active SSL, even if it's just self-signed. having to go externally exposed just to access things that are restricted without SSL is counter productive.

& a quick look at your compose links like if used access internally will not be permitted at all. Which also defats half the purpose of having in a home lab to start with. there really needs to be a way to enable internal Self-signed credentials too, like the OP was asking about

**Comment by iambibhas** [comment]
Just came across this issue. It should be in the documentation somewhere. Can't seem to find it.
